aiodocker/docker.py

```diff
@@ -320,6 +320,30 @@ def __init__(self, docker):
         )
         return DockerContainer(self.docker, id=data['Id'])
 
+    async def run(self, config, name=None):
+
+        try:
+            container = await self.create(config, name)
+        except DockerError as e:
+
+            # image not find, try pull it
+
+            if e.status == 404 and e.message.startswith("No such image:"):
```

Comment:
Is there any better way to check this case? I mean using string in error message is I bit fragile for library. What is payload in this case? I suspect payload has json information, it can be used for better error checking.